src/datasets/features/features.py

```diff
@@ -306,7 +306,10 @@ def encode_example(self, value):
         elif pa.types.is_floating(self.pa_type):
             return float(value)
         elif pa.types.is_string(self.pa_type):
-            return str(value)
+            if isinstance(value, str):
```

Comment:
just out of curiosity - this won't break anything where *e.g.* an int value is passed but assumed to be of type string?

Reply:
I think we have to discuss this indeed. I'm ok with accepting integers and floats to avoid breaking some scripts

Reply:
ok maybe just not allow lists then?

Reply:
In general, I think it's a pretty weird behavior from python that `str([1,2,3])` is allowed